In [36]:
import pandas as pd

from numpy import array

In [37]:
# read in dataset
xl = pd.ExcelFile("data/130N_Cycles_1-47.xlsx")
df = xl.parse("Specimen_RawData_1")
df

Time     Load
0           0.0000  0.06729
1           0.0018  0.07128
2           0.1000  0.05453
3           0.2000  0.00621
4           0.3000  0.00352
5           0.4000  0.02063
6           0.5000 -0.00168
7           0.6000 -0.01183
8           0.7000 -0.00167
9           0.8000 -0.00656
10          0.9000 -0.02108
11          1.0000 -0.03623
12          1.1000 -0.02316
13          1.2000 -0.05095
14          1.3000 -0.07705
15          1.4000 -0.05499
16          1.5000 -0.04504
17          1.6000 -0.06794
18          1.7000 -0.10153
19          1.8000 -0.05887
20          1.9000 -0.06541
21          2.0000 -0.07291
22          2.1000 -0.03818
23          2.2000 -0.04398
24          2.3000  0.00392
25          2.4000 -0.02976
26          2.5000  0.01042
27          2.6000  0.02127
28          2.7000 -0.01094
29          2.8000  0.00993
...            ...      ...
170071  16768.0000 -0.13978
170072  16768.1000 -0.12339
170073  16768.1990 -0.15841
170074  16768.2990 -0.12322
170075  16768.4000 -0.11189
170076  16768.5000 -0.12804
170077  16768.6000 -0.11056
170078  16768.6990 -0.09291
170079  16768.7990 -0.08599
170080  16768.9000 -0.10398
170081  16769.0000 -0.08131
170082  16769.1000 -0.05598
170083  16769.1990 -0.07377
170084  16769.2990 -0.09031
170085  16769.4000 -0.04457
170086  16769.5000 -0.03088
170087  16769.6000 -0.03074
170088  16769.6990 -0.04886
170089  16769.7990 -0.01057
170090  16769.9000 -0.03553
170091  16770.0000 -0.01936
170092  16770.1000  0.00220
170093  16770.1990  0.02822
170094  16770.2990  0.02014
170095  16770.4000  0.02088
170096  16770.5000  0.05239
170097  16770.6000  0.03399
170098  16770.6990  0.05024
170099  16770.7990  0.06006
170100  16770.9000  0.09558

[170101 rows x 2 columns]

This is what the dataset currently looks like - it has 170,101 rows and two columns.<br><br>
The dataset contains data from 47 cycles following an experiment. The output of these experiments form the two columns:<br>
- time (seconds)
- load (exerted force, in Newtons)
<br><br>
My task is to predict the load for cycles 48, 49, and 50.<br><br>
I will:
- derive the time for each cycle
- heating time for each cycle
- cooling time for each cycle
- number of cycles (I can do this my finding the local maxima and minima throughout the data)


In [38]:
# append data from time column to list
time = []
for item in df.index:
    time.append(df["Time"][item])

# append data from load column to list
load = []
for item in df.index:
    load.append(df["Load"][item])

In [40]:
# convert time array to np array for further processing
np_time = array(time)